plugins/tags/plugin.cpp

```diff
@@ -552,15 +539,18 @@ namespace golos { namespace plugins { namespace tags {
     }
 
     DEFINE_API(tags_plugin, get_discussions_by_blog) {
-        CHECK_ARG_SIZE(1)
+        PLUGIN_API_VALIDATE_ARGS(
+            (discussion_query, query)
+        );
 
-        auto query = args.args->at(0).as<discussion_query>();
         query.prepare();
         query.validate();
-        FC_ASSERT(query.select_authors.size(), "Must get blogs for specific authors");
+        GOLOS_CHECK_PARAM(query.select_authors,
+            GOLOS_CHECK_VALUE(query.select_authors.size(), "Must get blogs for specific authors"));
 
         auto& db = pimpl->database();
-        FC_ASSERT(db.has_index<follow::feed_index>(), "Node is not running the follow plugin");
+        GOLOS_ASSERT(db.has_index<follow::feed_index>(), golos::unsupported_operation, 
```

Comment:
`golos::unsupported_operation` — actually, it's not operation in terms of node. it's api method

Reply:
What do you mean? a) in this place must throw unsupported_method, or b) exception class golos::unsupported_operation must be named golos::unsupported_method?

Reply:
`operation` in a blockchain is a method which make changes.
It should be in a `transaction` with a `signature` from owner. 

`get_discussions_by_blog()` is an API method, which return result, it isn't registered in block_log.

Reply:
I only highlight some inconsistency and have no ready solution.
Yes, both cases can be valid (note for b — it can be generalised to contain both method and operation). Afaik, we have separate exception for missing api/api method which thrown early in plugin searching step. Maybe it can be used too.
I prefer reuse "missing api" exception if it's possible, or create separate one for unsupported method/required plugin.

Reply:
Exception doesn't used at plugin searching step. In this case json_rpc filled error message with JSON_RPC_METHOD_NOT_FOUND error code.
`golos::unsupported_operation` exception is it exception, which thrown when business logic can't execute operation/method, because it is unsupported (obsolete, in next hardfork, or some other errors).
`golos::unsupported_operation` is a subclass of `golos::operation_exception` which is used to notify about errors related with the call of some method. In fact, now it is used for both operations and API methods.
Seems like `operation` in exceptions confilcts with term `operation` in blockchain.

Reply:
Yes, it's not obvious that they are different `operations`. I was sure `golos::unsupported_operation` related to blockchain term

Reply:
Added golos::unsupported_api_method